feat(Alert{type=danger}): Deprecate alert danger type

sharvit opened the pull request:
BREAKING CHANGE: Alert{type=danger} is deprecated, migrate to Alert{type=error}

fix #144

**What**:
Deprecating `<Alert type="danger" />`
The component will continue to function normally, you should only get a warning message in your console.

**Additional issues**:
In what version we should drop the support of the `danger` type?
I think we should update this pr so the warning message will contain the version it will get dropped.

[storybook](https://rawgit.com/sharvit/patternfly-react/storybook/feature/deprecate-alert-danger-type/index.html?selectedKind=Alert&selectedStory=Alert%20types&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

sharvit commented:
I also added more tests to cover all the helpers.
I used snapshot testing there, feel pretty neat 👍

jeff-phillips-18 commented:
The story for Alert 'with Danger' should be updated to 'with Error'.

I see that we never document the valid types in the storybook, we should add that (and exclude 'danger'.

The toast notification story uses its own array of alert types, can it use the Alert constants? Currently it only allows 'danger' not 'error'.

jeff-phillips-18 commented on src/components/Alert/Alert.js:
```diff
@@ -3,13 +3,15 @@ import React from 'react';
  */
 const Alert = ({ children, className, onDismiss, type, ...props }) => {
+  warnIfDeprecatedType(type);
+
```
Should we auto-change the type from 'danger' to 'error' here?

sharvit replied:
I believe not, the helpers should know how to translate `danger` into `className` and `icon`

sharvit commented:
Thanks @jeff-phillips-18, i addressed all your issues.

I added another constant called `DEPRECATED_ALERT_TYPES` and removed the `danger` from the `ALERT_TYPES`

[storybook](https://rawgit.com/sharvit/patternfly-react/storybook/feature/deprecate-alert-danger-type/index.html?selectedKind=Alert&selectedStory=Alert%20types&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

priley86 commented:
very nice @sharvit ;)
